hash/blake2b.go

```diff
@@ -0,0 +1,75 @@
+// Go Substrate RPC Client (GSRPC) provides APIs and types around Polkadot and any Substrate-based chain RPC calls
+//
+// Copyright 2019 Centrifuge GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package hash
+
+import (
+	"hash"
+
+	"golang.org/x/crypto/blake2b"
+)
+
+type blake2b128Concat struct {
+	data []byte
```

Comment:
it would better to have `golang.org/x/crypto/blake2b` as part of the struct instead of `data`. Then our `Write` and `Sum` would just be proxy to the `golang.org/x/crypto/blake2b`